modules/sand/src/test/ts/browser/HtmlElementTest.ts

```diff
@@ -1,11 +1,53 @@
-import { assert, UnitTest } from '@ephox/bedrock-client';
-import { SandHTMLElement } from 'ephox/sand/api/Main';
-
-UnitTest.test('HtmlElementTest', () => {
-  const span = document.createElement('div');
-  assert.eq(false, SandHTMLElement.isPrototypeOf(null));
-  assert.eq(false, SandHTMLElement.isPrototypeOf(undefined));
-  assert.eq(false, SandHTMLElement.isPrototypeOf('a string'));
-  assert.eq(false, SandHTMLElement.isPrototypeOf({}));
-  assert.eq(true, SandHTMLElement.isPrototypeOf(span));
+import { context, describe, it } from '@ephox/bedrock-client';
+import { assert } from 'chai';
+
+import * as SandHTMLElement from 'ephox/sand/api/SandHTMLElement';
+
+describe('HtmlElementTest', () => {
+  context('isPrototypeOf', () => {
+    it('non elements should be false', () => {
+      assert.isFalse(SandHTMLElement.isPrototypeOf(null));
+      assert.isFalse(SandHTMLElement.isPrototypeOf(undefined));
+      assert.isFalse(SandHTMLElement.isPrototypeOf('a string'));
+      assert.isFalse(SandHTMLElement.isPrototypeOf({}));
+    });
+
+    it('nodes should be false', () => {
+      const text = document.createTextNode('text');
+      const comment = document.createComment('comment');
+      const frag = document.createDocumentFragment();
+      assert.isFalse(SandHTMLElement.isPrototypeOf(text));
+      assert.isFalse(SandHTMLElement.isPrototypeOf(comment));
+      assert.isFalse(SandHTMLElement.isPrototypeOf(frag));
+    });
+
+    it('SVG elements should be false', () => {
+      const svg = document.createElementNS('http://www.w3.org/2000/svg', 'svg');
+      assert.isFalse(SandHTMLElement.isPrototypeOf(svg));
+    });
+
+    it('same window elements should be true', () => {
+      const div = document.createElement('div');
+      assert.isTrue(SandHTMLElement.isPrototypeOf(div));
+    });
+
+    it('TINY-7374: different window elements should be true', () => {
+      const span = document.createElement('span');      // HTMLSpanElement
+      const strong = document.createElement('strong');  // HTMLElement
+      const iframe = document.createElement('iframe');
+      document.body.appendChild(iframe);
+
+      const iframeDoc = iframe.contentDocument;
+      iframeDoc.open();
+      iframeDoc.write('<html><body></body></html>');
+      iframeDoc.close();
+
+      iframeDoc.body.appendChild(span);
+      iframeDoc.body.appendChild(strong);
+      assert.isTrue(SandHTMLElement.isPrototypeOf(span));
+      assert.isTrue(SandHTMLElement.isPrototypeOf(strong));
```

Comment:
Is this asserting what we want it to? The `span` and `strong` were still created in the parent document. Would it be better to do something like

```typescript
const span = iframeDoc.createElement('span');
const strong = iframeDoc.createElement('strong');

assert.isTrue(SandHTMLElement.isPrototypeOf(span));
assert.isTrue(SandHTMLElement.isPrototypeOf(strong));
```

Reply:
No... that's the exact opposite of what we want to check and would have passed before this change. As it gets the owner document -> owner window -> HTML element currently so that would already work for the above. The case it fails is the one in a test, whereby it's created with a document in window x and then added to window y. Since it'll use the `HTMLElement` from window y it fails since the element prototype is from window x.